### ISSUE TITLE
Followed prow/getting_started_deploy.md, but stuck at "ContainerCreating" step

### ISSUE THREAD

**Issue by sunmou99** [kubernetes/test-infra#18856]
**What happened**: 
I was following the "Manual deployment". At the step "Add the prow components to the cluster" - "starter-gcs.yaml", I got STATUS:
NAME                                    READY   STATUS              RESTARTS   AGE
crier-8698b4f547-ms5px                  0/1     ContainerCreating   0          6m57s
deck-5f455fdc85-89mmw                   0/1     ContainerCreating   0          7m3s
deck-5f455fdc85-k5tq5                   0/1     ContainerCreating   0          7m2s
ghproxy-fd78f4c64-r9659                 1/1     Running             0          6m58s
hook-67f8665d79-lxgj5                   1/1     Running             0          7m
hook-67f8665d79-xnn52                   1/1     Running             0          7m
horologium-65bd7cfdbf-ks7kj             1/1     Running             0          7m1s
prow-controller-manager-f447f7b-sjpb5   1/1     Running             0          6m57s
sinker-546d7787f9-vrhf7                 1/1     Running             0          7m1s
statusreconciler-749969fdcb-nhztq       0/1     ContainerCreating   0          6m57s
tide-6477dd94bb-pgcbp                   0/1     ContainerCreating   0          6m58s

Check the Kubernetes Console, get the error msg: "**Does not have minimum availability**": 

**What you expected to happen**: 
STATUS should be running. 

**How to reproduce it (as minimally and precisely as possible)**: Just following the prow/getting_started_deploy.md and used "starter-gcs.yaml".

**Please provide links to example occurrences, if any**: 
https://pantheon.corp.google.com/kubernetes/workload?organizationId=433637338589&project=firebase-sdk-startup-time&pageState=(%22savedViews%22:(%22i%22:%2213ed370711df447e8bb118affc86a123%22,%22c%22:%5B%5D,%22n%22:%5B%5D))

**Anything else we need to know?**:
1. information from "kubectl cluster-info dump"
Failed loading config; disabling tracing: open /export/hda3/trace_data/trace_config.proto: no such file or directory

2. 
I noticed another error, though it's not likely related to the "ContainerCreating" error.
prow shows error msg: error while evaluating the ingress spec: service "prow/deck" is type "ClusterIP", expected "NodePort" or "LoadBalancer"; service "prow/hook" is type "ClusterIP", expected "NodePort" or "LoadBalancer"; service "prow/deck" is type "ClusterIP", expected "NodePort" or "LoadBalancer".
I added node type NodePort, but still not working. The error msg turn into "Some backend services are in UNKNOWN state"
https://pantheon.corp.google.com/kubernetes/discovery?organizationId=433637338589&project=firebase-sdk-startup-time&pageState=(%22savedViews%22:(%22i%22:%22ec34bba4525345e5970ed4081b18bfbb%22,%22c%22:%5B%5D,%22n%22:%5B%5D))

**Comment by sunmou99** [comment]
> I added node type ClusterIP, but still not working. The error msg turn into "Some backend services are in UNKNOWN state"

Sorry, I mean "I added node type **NodePort**, but still not working".

**Comment by deedubs** [comment]
starter-gcs.yaml doesn't create the secret `gcs-credentials` for you.  You need to do that yourself.

1. Create a service account to run prow under

**Comment by sunmou99** [comment]
> 3\. kubectl create secret generic gcs-credentials --from-file=gcs-credentials.json -n prow

Thanks @deedubs . Now most workloads running except tide:
NAME                                       READY   STATUS             RESTARTS   AGE
crier-6ff8d884db-c7wmj                     1/1     Running            0          2m28s
deck-86799fbf6b-474l5                      0/1     Running            0          2m30s
deck-86799fbf6b-hvltn                      0/1     Running            0          2m31s
ghproxy-75c45cd949-gf9fg                   1/1     Running            0          2m29s
hook-85f4754c7d-49zvm                      1/1     Running            0          2m29s
hook-85f4754c7d-svpgw                      1/1     Running            0          2m29s
horologium-66566cdf4d-wvzpn                1/1     Running            0          2m30s
prow-controller-manager-6b54b68b8c-d2cnb   1/1     Running            0          2m29s
sinker-59786df698-w5sz2                    1/1     Running            0          2m30s
statusreconciler-9d756b865-m29bm           1/1     Running            0          2m29s
**tide-5bc8cf4c99-2wnsk**                      0/1     **CrashLoopBackOff**   4          2m29s

Error Msg:
Error creating: pods "tide-5bc8cf4c99-" is forbidden: error looking up service account prow/tide: serviceaccount "tide" not found | FailedCreate

**Comment by roshanirathi** [comment]
I am facing the same issue with tide. Any solution for this?

> > 3. kubectl create secret generic gcs-credentials --from-file=gcs-credentials.json -n prow
> 
> Thanks @deedubs . Now most workloads running except tide:
> NAME READY STATUS RESTARTS AGE
> crier-6ff8d884db-hpjj4 1/1 Running 0 19m
> deck-86799fbf6b-mnxxw 1/1 Running 0 19m
> deck-86799fbf6b-n9w4v 1/1 Running 0 19m
> ghproxy-75c45cd949-vbnz2 1/1 Running 0 19m
> hook-85f4754c7d-nnk64 1/1 Running 0 19m
> hook-85f4754c7d-rhk2x 1/1 Running 0 19m
> horologium-66566cdf4d-bdpfs 1/1 Running 0 19m
> prow-controller-manager-6b54b68b8c-c97zk 1/1 Running 0 19m
> sinker-59786df698-27zhl 1/1 Running 0 19m
> statusreconciler-9d756b865-brpl6 1/1 Running 0 19m
> tide-5bc8cf4c99-mfh2d 0/1 CrashLoopBackOff 8 19m
> 
> Here are some logs:
> [tide_describe.txt](https://github.com/kubernetes/test-infra/files/5144552/tide_describe.txt)
> [tide_log.txt](https://github.com/kubernetes/test-infra/files/5144553/tide_log.txt)

**Comment by simon-wessel** [comment]
I ran into the same two problems and both are connected to the documentation:

1)
As @deedubs correctly states the command to add the secret is:

`kubectl create secret generic gcs-credentials --from-file=gcs-credentials.json -n prow`

The command in the documentation uses the "test-pods" namespace for the secret though. That is why some pods are missing the credentials and won't come up.

2)
The reason why tide will not start when strictly following the Getting Started guide is that in 7fa77f6f31c8d955ed7b0487f97dde4ae0b06d55 the arguments `--status-path=gs://tide/tide-status` and `--history-uri=gs://tide/tide-history.json` were added to the default `starter-gcs.yaml`. They add optional persistent action histories and refer to a bucket with the name `tide` that is not accessible to your Google Cloud user and may have billing disabled. That is why you get the 403 response with the strange error "The billing account for the owning project is disabled in state absent" even though everything may be fine with your billing.

There are two solutions to the second issue:

- Remove the two lines for `--status-path` and `--history-uri` from your `starter-gcs.yaml` to disable persistent action histories (commenting them out with will do)
- Create/Use a bucket that is accessible to your Google Cloud user as is documented in [Persistant Storage of Action History](https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/config.md#persistent-storage-of-action-history) and change the name in your `starter-gcs.yaml`

I can create a PR for the documentation changes tomorrow.
